drivers/at86rf2xx/at86rf2xx_rf_ops.c

```diff
@@ -0,0 +1,971 @@
+}
+
+static int _request_on(ieee802154_dev_t *hal)
+{
+    DEBUG("at86rf2xx_rf_ops: request_on\n");
+    at86rf2xx_t *dev = hal->priv;
+    mutex_lock(&dev->lock);
+#if !AT86RF2XX_IS_PERIPH
+    gpio_irq_enable(dev->params.int_pin);
+#endif
+    /* enable interrupts */
+    at86rf2xx_reg_write(dev, AT86RF2XX_REG__IRQ_MASK,
+                        AT86RF2XX_IRQ_STATUS_MASK__TRX_END | AT86RF2XX_IRQ_STATUS_MASK__RX_START);
+
+
+#if AT86RF2XX_IS_PERIPH
+    /* Setting SLPTR bit in TRXPR to 0 returns the radio transceiver
+     * to the TRX_OFF state */
+    *AT86RF2XX_REG__TRXPR &= ~(AT86RF2XX_TRXPR_SLPTR);
+#else
+    gpio_clear(dev->params.sleep_pin);
+#endif
+
+    mutex_unlock(&dev->lock);
+    return 0;
+}
+
+static int _confirm_on(ieee802154_dev_t *hal)
+{
+    DEBUG("at86rf2xx_rf_ops: confirm_on\n");
+    at86rf2xx_t *dev = hal->priv;
+    mutex_lock(&dev->lock);
+    int status = at86rf2xx_reg_read(dev, AT86RF2XX_REG__TRX_STATUS)
+                     & AT86RF2XX_TRX_STATUS_MASK__TRX_STATUS;
+
+     if (status != AT86RF2XX_TRX_STATUS__TRX_OFF)
+     {
+         mutex_unlock(&dev->lock);
+         return -EAGAIN;
+     }
+    at86rf2xx_reg_read(dev, AT86RF2XX_REG__IRQ_STATUS);
+    mutex_unlock(&dev->lock);
+    return 0;
+}
+
+static int _len(ieee802154_dev_t *hal)
+{
+    DEBUG("at86rf2xx_rf_ops: len\n");
+    at86rf2xx_t *dev = hal->priv;
+    mutex_lock(&dev->lock);
+    uint8_t phr = at86rf2xx_peek_rx_len(dev);
+    mutex_unlock(&dev->lock);
+    return (phr & 0x7f) - 2;
+}
+
+static int _off(ieee802154_dev_t *hal)
+{
+    DEBUG("at86rf2xx_rf_ops: off\n");
+    at86rf2xx_t *dev = hal->priv;
+    mutex_lock(&dev->lock);
+    at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_STATE, AT86RF2XX_STATE_FORCE_TRX_OFF);
+
+    /* Discard all IRQ flags, framebuffer is lost anyway */
+    at86rf2xx_reg_read(dev, AT86RF2XX_REG__IRQ_STATUS);
+    /* Go to SLEEP mode from TRX_OFF */
+#if AT86RF2XX_IS_PERIPH
+    /* reset interrupts states in device */
+    /* Setting SLPTR bit brings radio transceiver to sleep in in TRX_OFF*/
+    *AT86RF2XX_REG__TRXPR |= (AT86RF2XX_TRXPR_SLPTR);
+#else
+    gpio_irq_disable(dev->params.int_pin);
+    gpio_set(dev->params.sleep_pin);
+#endif
+    mutex_unlock(&dev->lock);
+    return 0;
+}
+static int _request_set_rx(at86rf2xx_t *dev)
+{
+    DEBUG("at86r2xx_rf_ops: request_set_rx\n");
+    at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_STATE, AT86RF2XX_PHY_STATE_RX);
+    return 0;
+}
+
+static int _request_set_tx(at86rf2xx_t *dev, bool force)
+{
+    DEBUG("at86r2xx_rf_ops: request_set_tx\n");
+    /* Prevent issue described in https://github.com/RIOT-OS/RIOT/pull/11256 */
+    if (at86rf2xx_get_status(dev) == AT86RF2XX_PHY_STATE_RX_BUSY && !force) {
+        return -EBUSY;
+    }
+    at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_STATE, AT86RF2XX_TRX_STATE__FORCE_PLL_ON);
+    at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_STATE, AT86RF2XX_PHY_STATE_TX);
+    return 0;
+}
+
+static int _request_cca(at86rf2xx_t *dev)
```

Comment:
`NETOPT_IS_CHANNEL_CLR` is no longer available, which previously triggered `at86rf2xx_cca()`.
Should it now trigger `_request_cca()`?

Reply:
> We still have bool at86rf2xx_cca(at86rf2xx_t *dev) I think you did not reuse it here because of the request/confirm pattern. If there is no reason to keep both, we could remove/deprecate the old function.

I just forgot to deprecate it.

> NETOPT_IS_CHANNEL_CLR is no longer available, which previously triggered at86rf2xx_cca().
Should it now trigger _request_cca()?

Yes, we should implement that now in `netdev_ieee802154_submac/netdev_ieee802154_submac.c`.